### PR TITLE
Fixes #11887

### LINKED ISSUE #11887: adjust searcher heap usage
It would be nice to change the max heap usage of the searcher in memory constrained environments like a cluster login node (which was killing helix when searching through large projects)

I posted [this](https://www.reddit.com/r/HelixEditor/comments/1g2aa6n/have_to_switch_back_to_nvim_for_memory_usage_on/) on the helix subreddit, and one of the commenters solved my issue and suggested I make a feature request. This is the patch that fixed my issue:

```
diff --git a/helix-term/src/commands.rs b/helix-term/src/commands.rs
index f25e8254..ecfcacfc 100644
--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -2327,6 +2327,7 @@ fn format(&self, current_path: &Self::Data) -> Row {

                 std::thread::spawn(move || {
                     let searcher = SearcherBuilder::new()
+                        .heap_limit(Some(5*1024*1024))
                         .binary_detection(BinaryDetection::quit(b'\x00'))
                         .build();
```

I have never written rust but if someone could point me in the general direction of how to add an options to the `config.toml` file and pass it to the searcher builder I would be be willing to try and add this as a feature.

### PR DESCRIPTION
**Pull request by mjkpolo**
Allow changing the heap memory used for memory constrained environments. Specifying in MB for simplicity.

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -2423,6 +2423,13 @@ fn global_search(cx: &mut Context) {
         let injector = injector.clone();
         async move {
             let searcher = SearcherBuilder::new()
+                .heap_limit(
+                    if let Some(max_heap_usage) = config.file_picker_config.max_heap_usage {
+                        Some(max_heap_usage * 1024 * 1024)
+                    } else {
+                        None
+                    }
```
Comment (poliorcetics) [inline helix-term/src/commands.rs:2431]:
```suggestion
                    config.file_picker_config.max_heap_usage.map(|m| max_heap_usage * 1024 * 1024)
```

**Comment by RoloEdits** [comment]
I left a review but I don't see my name as a reviewer at the top right, did github eat my review?
image

**Comment by poliorcetics** [comment]
> I left a review but I don't see my name as a reviewer at the top right, did github eat my review?

It's marked as "pending", you've not posted it

**Comment by poliorcetics** [comment]
Also @mjkpolo you should modify the book to add the new option :)

**Comment by mjkpolo** [comment]
Actually maybe this should be closed, this fixed an issue with 24.07, but I can't replicate it with the new searcher. If I really stress test it with my whole home directory, it eventually runs out of memory with and without this option set

**Comment by RoloEdits** [comment]
Yeah, the new global search was made in mind with large codebases. If you noticed no improvement with changes here, then it should be fine if you want to close it. The other option would be to leave it open, and after the next release, if anyone has memory problems, they can see if this works for them. You might get pinged in here to rebase on master if so, though, so it could become a maintenance burden for you, at least for the first few months its out. After that should be fine to close it.

Perhaps you can further document the environment you were having problems with so there is a baseline?

Either way, the changes would get documented here if even if you close it, so they could do it themselves later, if needed.

**Comment by mjkpolo** [comment]
Sure, I am good leaving it open until the next release and then closing it.
The environment I was using was just searching through [llama.cpp](https://github.com/ggerganov/llama.cpp) on the [AMD HPC](https://amdresearch.github.io/hpcfund/hardware.html) login node. It was probably large trace files in that directory causing memory issues for search though, or models, since llama.cpp isn't a massive project.
If I play with it more and figure out how heap-limit helps I can add to this

**Pull request closed without merging**